querydsl-core/src/main/java/com/querydsl/core/types/dsl/StringExpression.java

```diff
@@ -149,8 +151,9 @@ public BooleanExpression contains(Expression<String> str) {
      * @return this.contains(str)
      * @see java.lang.String#contains(CharSequence)
      */
-    public BooleanExpression contains(String str) {
-        return contains(ConstantImpl.create(str));
+    public BooleanExpression contains(@NotNull String str) {
+        // It can return null or throw an exception.
+        str == null ? null : contains(ConstantImpl.create(str));
```

Comment:
This is not a fix but just moves the problem to another point.

There is no proper way of implementing null handling here, because SQL has no concept of Java's null.  The NullPointerException is intentional here, because the method is not supposed to be called with null.

Therefore we will not accept this PR.